base/irrationals.jl

```diff
@@ -221,6 +221,15 @@ end
 log(::Irrational{:e}) = 1 # use 1 to correctly promote expressions like log(x)/log(e)
 log(::Irrational{:e}, x::Number) = log(x)
 
+# These methods should go to special/trig.jl, but Irrational is not yet defined when the
+# file is loaded.
+sinpi(x::Irrational) = sinpi(float(x))
+cospi(x::Irrational) = cospi(float(x))
+sinc(x::Irrational) = sinc(float(x))
+cosc(x::Irrational) = cosc(float(x))
+sinc(z::Complex{<:Irrational}) = sinc(float(z))
+cosc(z::Complex{<:Irrational}) = cosc(float(z))
```

Comment:
Yes, there are methods for `sinpi` and `cospi` in MPFR.  You suggest something like this?
```julia
for c in (e, pi, catalan, golden)
    @eval begin
        Base.sinpi(x::typeof($c)) = $(Float64(sinpi(big(c))))
        Base.cospi(x::typeof($c)) = $(Float64(cospi(big(c))))
        Base.sinc(x::typeof($c)) = $(Float64(sinc(big(c))))
        Base.cosc(x::typeof($c)) = $(Float64(cosc(big(c))))
        Base.sinc(z::typeof(Complex($c, $c))) = $(Complex{Float64}(sinc(big(Complex(c, c)))))
        Base.cosc(z::typeof(Complex($c, $c))) = $(Complex{Float64}(cosc(big(Complex(c, c)))))
    end
end
```
I also thought about this, but the same can apply to all other functions and I don't think we're going to define a method for all functions in base library with all irrational numbers.  These definitions are anyway required to make custom irrational numbers work (otherwise users will need to define the methods themselves).  In any case, these methods should be more accurate than `function(pi * x)` (I compared with the bigfloat implementations), which is the goal of `*pi` functions.